drivers/adc/adc_stm32.c

```diff
@@ -735,28 +746,21 @@ static int start_read(const struct device *dev,
 	defined(CONFIG_SOC_SERIES_STM32G0X) || \
 	defined(CONFIG_SOC_SERIES_STM32G4X) || \
 	defined(CONFIG_SOC_SERIES_STM32H7X)
-	LL_ADC_EnableIT_EOC(adc);
+	if (data->channel_count > 1) {
+		LL_ADC_EnableIT_EOS(adc);
+	}
+	else {
+		LL_ADC_EnableIT_EOC(adc);
+	}
 #elif defined(CONFIG_SOC_SERIES_STM32F1X)
 	LL_ADC_EnableIT_EOS(adc);
 #else
 	LL_ADC_EnableIT_EOCS(adc);
 #endif
 
-#ifdef CONFIG_ADC_STM32_DMA
-	data->dma_error = 0;
-	adc_stm32_dma_start(dev, sequence);
-#endif
-
 	adc_context_start_read(&data->ctx, sequence);
 
-	int result = adc_context_wait_for_completion(&data->ctx);
-
-#ifdef CONFIG_ADC_STM32_DMA
-	dma_stop(data->dma.dma_dev, data->dma.channel);
-	result = (data->dma_error ? data->dma_error : result);
```

Comment:
It seems that the `dma_error` flag is now never used? Is this deliberate?

Reply:
Not really, but I can't find a spot where I can place it, since it isn't start related anymore but started/stopped upon every callback. If you know of a spot where to fit it in let me know.

Reply:
Do we actually have a situation, when DMA is setup without an error, and error is returned during conversion?

Reply:
@aborowski-msol I think we should ideally have this check. The DMA might break for many reasons which the DMA driver will catch. The most likely is probably incompatable DTS settings.There's also a list of things that can go wrong in the datasheet.

@JvanDooren I think for the synchronous case it can remain as Adam did it in `start_read`:
```
	int result = adc_context_wait_for_completion(&data->ctx);

#ifdef CONFIG_ADC_STM32_DMA
	dma_stop(data->dma.dma_dev, data->dma.channel);
	result = (data->dma_error ? data->dma_error : result);
#endif

	return result;
```
As for the asynchronous case I think the information is lost since it's not possible to put it as part of the callback. Although, if we modify `adc_context.c` we could add it to the existing `status` flag. It's passed to `k_poll_signal_raise` when the async read is done. This wasn't used before, so that could be why it doesn't exist at the moment. Not sure if should go that far though.

Reply:
Ok, I'll add it again to the sync read, but the async read is actually a different beast. It's actually weird that there is no status in the callback parameter list, nor is there a get_status method in the api. The status field in the context is for internal use, I don't dare to touch that.